src/ruleset.rs

```diff
@@ -551,6 +553,34 @@ pub trait RulesetCreatedAttr: Sized + AsMut<RulesetCreated> + Compatible {
         <Self as AsMut<RulesetCreated>>::as_mut(&mut self).no_new_privs = no_new_privs;
         self
     }
+
+    /// Creates a copy of this object and duplicates the file descriptor with
+    /// CLOEXEC set. The `RulesetRef` object that is produced is intended
+    /// to be a readonly copy of the `RulesetCreated` object, however this
+    /// is not currently supported by the landlock kernel module. The intention
+    /// behind this method is to provide a way to keep a parent ruleset intact so
+    /// it can be applied to any number of new processe or threads without
+    /// creating the object over and over.
+    ///
+    /// Any rule changes applied to the Rulescreated object that will be inherited
+    /// by the RulesetRef object.
+    ///
+    /// On error, returns a wrapped [`RulesetError::NewRef`].
+    fn new_ref(&self) -> Result<RulesetCreated, RulesetError> {
+        let self_ref = self.as_ref();
+
+        let dup_fd = unsafe { libc::fcntl(self_ref.fd, libc::F_DUPFD_CLOEXEC) };
+        if dup_fd == -1 {
+            return Err(RulesetError::NewRef(Error::last_os_error()));
+        }
+
+        Ok(RulesetCreated {
```

Comment:
You can use `RulesetCreated::new()` instead, wrap this new ruleset into a `RulesetRef` object, and implement a `restrict_self()` method for `RulesetRef`. This way, no new trait and fully compatible with existing code. The documentation of this new `restrict_self()` method should just point to the `RulesetCreated::restrict_self()` one.